### PR TITLE
go: add deoplete#sources#go#cgo_only config for disable go completion

### PR DESCRIPTION
**Pull request by zchee**
Add `deoplete#sources#go#cgo_only` config for disable go completion.

It useful for users want to use both gopls based lsp and deoplete-go.

### CODE REVIEW

**Comment by Shougo** [comment]
Thank you!
It is merge-able?

**Comment by zchee** [comment]
@Shougo yep, already ready to merge.

---

BTW, in Go community, we has been stopped using gocode, we using gopls (Go core team maintains lsp server for Go). I think you already know, that's features are not only of completion. diag, lint and etc.

So, normally deoplete-go will be an archive project. But I want to continue to maintain. Bacause of there is no lsp server tht's support cgo completion with libclang at least now.

In other words, I planning to reinvent deoplete-go and make it a deoplete sources specialized for cgo completion in the future.
Also in addition, fortunately libclang's python binding has officially started to support python3.
What do you think about it?

**Comment by zchee** [comment]
ah, forgot one thing.

FYI, I was already pushed libclang v9.0.0 support branch (but still WIP)

https://github.com/deoplete-plugins/deoplete-go/tree/libclang/v9.0.0

**Comment by Shougo** [comment]
OK.  Merged.

> FYI, I was already pushed libclang v9.0.0 support branch (but still WIP)

I will merge it if it is completed.

> So, normally deoplete-go will be an archive project. But I want to continue to maintain. Bacause of there is no lsp server tht's support cgo completion with libclang at least now.

Hm.  OK.  I think it should be documented.

> In other words, I planning to reinvent deoplete-go and make it a deoplete sources specialized for cgo completion in the future.

It seems nice.  deoplete-cgo?

> Also in addition, fortunately libclang's python binding has officially started to support python3.

Python3 is better.  Python2 is deprecated.

**Comment by Shougo** [comment]
Ah, please add `deoplete#sources#go#cgo_only` documentation.

**Comment by zchee** [comment]
@Shougo I see, will do (maybe tonight)
